sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -146,6 +147,18 @@ object RapidsPluginUtils extends Logging {
     }
     // set driver timezone
     conf.set(RapidsConf.DRIVER_TIMEZONE.key, ZoneId.systemDefault().normalized().toString)
+
+    // If spark.rapids.sql.multiThreadedRead.numThreads is not set explicitly, or it is set
+    // explicitly as value 0, then we derive it from other settings. Otherwise, we keep the
+    // users' setting.
+    val numThreadsKey = RapidsConf.MULTITHREAD_READ_NUM_THREADS.key
+    if (!conf.contains(numThreadsKey) || conf.get(numThreadsKey).toInt == 0) {
```

Comment:
my comment about setting it to 0 seems to have been addressed by checking `.conf.contains` but now checking for 0 here is weird and raises an issue.

If the user sets this value to 0 and `EXECUTOR_CORES_KEY` isn't set, what happens? Do we crash or do we hang?

At this rate if at line 161 of this file `numThreadsKey` is 0, we should probably throw if we are not doing so already when creating the multi threaded reader.

Reply:
There's constraint that can be added to the config entry (RapidsConf) `checkValue` that would handle this:

```
  conf("spark.rapids.sql.multiThreadedRead.numThreads")
      .checkValue(v => v > 0, "numThreads must be greater than 0")
      // + what you have now      
```

thoughts @tgravescs ?

Reply:
yeah that sounds like beter approach